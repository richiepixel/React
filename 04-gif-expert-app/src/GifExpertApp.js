import React, { useState } from 'react';
import AddCategory from './Components/AddCategory';
import GifGrid from './Components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
  // const [categories, setCategories] = useState(['Zelda']);
  const [categories, setCategories] = useState(defaultCategories);
  // const handleAdd = () => {
  //   setCategories((c) => [...categories, 'Chivas']);
  // };

  return (
    <>
      <h2>GiftExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
