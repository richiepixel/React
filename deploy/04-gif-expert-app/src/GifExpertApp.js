import React, { useState } from 'react';
import AddCategory from './Components/AddCategory';
import GifGrid from './Components/GifGrid';

const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Zelda']);

  // const handleAdd = () => {
  //   setCategories((c) => [...categories, 'Chivas']);
  // };

  return (
    <>
      <h2>GiftExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
