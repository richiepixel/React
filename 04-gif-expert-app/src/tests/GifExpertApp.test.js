import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en  <GifExpertApp />', () => {
  test('debe de mostrar el componente correctamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
    const categories = ['Zelda', 'Real Madrid'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
