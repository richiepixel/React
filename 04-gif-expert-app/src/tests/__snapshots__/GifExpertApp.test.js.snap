// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <GifExpertApp /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
